src/msg/async/Protocol.cc

```diff
@@ -1078,7 +1111,9 @@ void ProtocolV1::randomize_out_seq() {
 ssize_t ProtocolV1::write_message(Message *m, bufferlist &bl, bool more) {
   FUNCTRACE(cct);
   ceph_assert(connection->center->in_thread());
-  m->set_seq(++out_seq);
+  if (m->get_seq() == 0) {
```

Comment:
So if a message was being sent to OSD ```a``` with seq 1 but was then redirected to OSD ```b```, it would keep seq 1 even if the connection to OSD ```b``` had an unrelated in-flight seq 1 message?

Reply:
As in _prepare_osd_op, the resent message will be a new message with sequence 0.